libdistopia/src/distopia.cpp

```diff
@@ -685,6 +685,169 @@ namespace distopia {
             }
         }
 
+        template <typename V>
+        HWY_INLINE void LoadInterleavedIdx(const unsigned int *idx, const float *src,
```

Comment:
@hmacdope this is the interesting function that does the IDX gather here.

In general, the width of the integer part of the `GatherIndex`needs to match the width of the floating point type, so for the current signature of `int` indices, the float case is simple and the double case requires an upcase (see below).

If we wanted to have int64 indices in the signature for `Idx` functions, we'd probably need two more specialisations to this function, which isn't too much of a drama.  The int64 and float case is a little odd as arguably the ints could overflow, so maybe you couldn't vectorise this as hard... (using current highway bindings at least).